src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1639,19 +1637,17 @@ private TokenCacheItem findFamilyItemForUser(final UserIdentifierType userIdenti
             return null;
         }
 
-        Iterator<TokenCacheItem> allItems = mTokenCacheStore.getAll();
-        if (allItems == null || !allItems.hasNext()) {
+        Iterable<TokenCacheItem> allItems = mTokenCacheStore.getAll();
+        if (allItems == null || !allItems.iterator().hasNext()) {
```

Comment:
IMHO -  `ITokenCacheStore` need to have `hasTokes` or something like that. `ITokenCacheStore` should become just a collection on it's own.